exec/java-exec/src/main/java/org/apache/drill/exec/work/filter/RuntimeFilterRouter.java

```diff
@@ -303,10 +332,28 @@ private Wrapper findPhysicalOpContainer(Wrapper wrapper, PhysicalOperator op) {
         return opContainer;
       }
     }
-    //should not be here
-    throw new IllegalStateException(String.format("No valid Wrapper found for physicalOperator with id=%d", op.getOperatorId()));
+    return null;
+  }
+
+  private Wrapper findRuntimeFilterContainer(Wrapper wrapper, long runtimeFilterIdentifier) {
+    boolean contain = containsRuntimeFilterPhysicalOperator(wrapper, runtimeFilterIdentifier);
+    if (contain) {
+      return wrapper;
+    }
+    List<Wrapper> dependencies = wrapper.getFragmentDependencies();
+    if (CollectionUtils.isEmpty(dependencies)) {
+      return null;
+    }
+    for (Wrapper dependencyWrapper : dependencies) {
+      Wrapper opContainer = findRuntimeFilterContainer(dependencyWrapper, runtimeFilterIdentifier);
+      if (opContainer != null) {
+        return opContainer;
+      }
+    }
+    return null;
```

Comment:
`findRuntimeFilterContainer` and `findPhysicalOpContainer` are doing same thing except they are using different visitor. We can refactor this code as below then caller can call with appropriate visitor.
```suggestion
private Wrapper findPhysicalOpContainer(Wrapper wrapper, PhysicalVisitor visitor) {
       boolean contain = containsPhysicalOperator(visitor);
    if (contain) {
      return wrapper;
    }
    List<Wrapper> dependencies = wrapper.getFragmentDependencies();
    if (CollectionUtils.isEmpty(dependencies)) {
      return null;
    }
    for (Wrapper dependencyWrapper : dependencies) {
      Wrapper opContainer = findPhysicalOpContainer(dependencyWrapper, visitor);
      if (opContainer != null) {
        return opContainer;
      }
    }
    return null;
  }
```

Reply:
make sense.